StackExchange/Stacks
# feat(topbar): add popover child component

[STACKS-613](https://stackoverflow.atlassian.net/browse/STACKS-613)

This PR adds a Stacks topbar popover component so we can control the size and overflow behavior, primarily to meet the success criteria for [WCAG SC 1.4.0 Reflow](https://www.w3.org/WAI/WCAG22/Understanding/reflow.html). 

> [!NOTE]
> I still need to add tests to this PR, but I'm opening it for review to get eyes on it earlier rather than later.

## New classes

This PR introduces the following child component classes to the topbar:

- `s-topbar--searchbar--popover`: This class applies a few sensible defaults to the popover in this context (specifically it sets `padding` to `0` and min/max width)
- `.s-topbar--searchbar--popover-content`: This class is intended to wrap the content within the popover to ensure it can resize and scroll vertically (if necessary)

## Concessions

Since the topbar height is set with `--theme-topbar-height`, we know the height of the topbar for certain, but _we cannot know for sure_ what the height of the search group is when it's rendered below the topbar (in the case of small viewports). Because of this, I took an educated guess and yielded on the side of caution to ensure the popover is rendered within the viewport.

It would be unlikely consumers would manually alter the height of the search group in mobile in such a way that it's significantly larger than the default, potentially rendering a portion of the popover outside of the viewport boundaries. I think this is a safe approach but I'm willing to reconsider if objections arise.


## TODO

- [ ] Add tests specifically for the popover when it's shown
- [ ] **Before merging**: Remove excess popover content

## Screenshots

<details>
<summary>320 x 256</summary>

![20240523-052206](https://github.com/StackExchange/Stacks/assets/647177/bf6fc47d-7c0d-4509-8dd6-a74efbc4e1fc)

</details>

<details>
<summary>770 x 256</summary>


![20240523-052050](https://github.com/StackExchange/Stacks/assets/647177/edf8e78d-d07f-4724-8214-39f6da0477cf)

</details>

## To test

For ease of testing this PR, I've added the popover to the main topbar of the Stacks docs site. It's temporary and will be removed before merging this PR.

- Go to the [Stacks docs](https://deploy-preview-1739--stacks.netlify.app/)
- Focus the search input in the main topbar of the docs site
- Notice the addition of a popover anchored to the search input
- Make the browser real small and notice the popover resizes to stay within the boundries of the viewport and introduces vertical scrolling if needed.
  - **Tip**: [320 x 256](https://www.w3.org/WAI/WCAG22/Understanding/reflow.html) is the min supported viewport size

### Bonus if you're ambitious (I don't necessarily recommend it, though)

- In the [topbar docs page](https://deploy-preview-1739--stacks.netlify.app/product/components/topbar), test out some of the other topbars and their respective popovers
  - You'll notice some overlapping issues. Ignore that since it's a docs-only issue.
  - You'll also notice you need to scroll so the topbar you're interacting with is where a topbar _would_ be on the page (at the top, usually 😝) in order to see the expected behavior wrt height. This is why I included the popover in the main topbar. You're probably better off just testing it there.
